src/modules/sensors/camera.c

```diff
@@ -8,6 +8,7 @@
 #define CAMERA_NAME                 "Camera"
 #define CAMERA_ILL_MAX              255
 #define CAMERA_SUBSYSTEM            "video4linux"
+#define HISTOGRAM_STEPS             20
```

Comment:
Using 20 steps here gives us about 5% slices to average, I tested some other values and they all produce decent results, tweaking this just adjusts granularity.

Reply:
I'd make HISTOGRAM_STEPS overridable at compile time from CMakeLists.txt.

Reply:
I've made the other changes you suggested locally, but I'm afraid I have no idea how to drive CMake.

Reply:
That is a no-issue. I can do that myself, or I can guide you if you wish!